src/ui/projects/CreateProjectPage.js

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useState, useContext } from "react";
+import PropTypes from "prop-types";
+import ScrollToTop from "../router/ScrollToTop";
+import NavBar from "../navigation/NavBar";
+import {
+  ProjectGrid,
+  ProjectGridContainer,
+  ProjectGridHeader,
+  ProjectGridHeaderRow,
+  Filter,
+  Separator,
+  SearchInput,
+  ProjectGridContent,
+  ErrorMessage
+} from "./ProjectGrid";
+import Footer from "../navigation/Footer";
+import PrimaryLink from "../inputs/PrimaryLink";
+import { Button } from "../inputs/Button";
+import { ProjectsSection, ProjectsContainer, ProjectsHeader } from "./ProjectsPage";
+import { ApiContext } from "../contexts/ApiContext";
+import { Link } from "react-router-dom";
+import InfiniteScroll from "react-infinite-scroller";
+import usePaginatedSearch from "./usePaginatedSearch";
+
+export default function CreateProjectPage({ history, location }) {
+  const api = useContext(ApiContext);
+
+  const queryParams = new URLSearchParams(location.search);
+
+  const [params, setParams] = useState({
+    source: "scene_listings",
+    filter: queryParams.get("filter") || "featured-remixable",
+    q: queryParams.get("q") || ""
+  });
+
+  const updateParams = useCallback(
+    nextParams => {
+      const search = new URLSearchParams();
+
+      for (const name in nextParams) {
+        if (name === "source" || !nextParams[name]) {
+          continue;
+        }
+
+        search.set(name, nextParams[name]);
+      }
+
+      history.push(`/projects/create?${search}`);
+
+      setParams(nextParams);
+    },
+    [history]
+  );
+
+  const onChangeQuery = useCallback(
+    value => {
+      updateParams({
+        source: "scene_listings",
+        filter: "remixable",
+        q: value
+      });
+    },
+    [updateParams]
+  );
+
+  const onSetFeaturedRemixable = useCallback(() => {
+    updateParams({
+      ...params,
+      filter: "featured-remixable",
+      q: ""
+    });
+  }, [updateParams, params]);
+
+  const onSetAll = useCallback(() => {
+    updateParams({
+      ...params,
+      filter: "remixable",
+      q: ""
+    });
+  }, [updateParams, params]);
+
+  const onSelectScene = useCallback(
+    scene => {
+      const search = new URLSearchParams();
+      search.set("sceneId", scene.id);
+      history.push(`/projects/new?${search}`);
+    },
+    [history]
+  );
+
+  const { loading, error, entries, hasMore, loadMore } = usePaginatedSearch(
+    `${api.apiURL}/api/v1/media/search`,
+    params
+  );
+
+  const filteredEntries = entries.map(result => ({
```

Comment:
Based on the name I was expecting this to filter out some results. Looks like its just cleaning up the response a bit. Might make sense to do this upstream somewhere, maybe in api.js (though as written `usePaginatedSearch` doesn't end up going through api.js so that would need to be reworked also).

Reply:
The problem is that projects and the media search API return different shaped responses. So this just maps scenes into something shaped like a project. Once we have a third type of media that uses this grid (I think that'll come soon) I'll refactor this and make the grid more generic.